### PR TITLE
Adds UX for connecting to dapps via the connected sites screen

### PR DESCRIPTION
**Pull request by danjm**
Demo video: https://streamable.com/ulah5

The PR adds the following feature: if a user has a non-metamask tab in focus, that has not granted `eth_accounts` permissions to MetaMask,  and goes to the "Connect Sites" screen, they will be able to click a button to connect to that focused/active tab.

### CODE REVIEW

File: ui/app/components/app/connected-sites-list/connected-sites-list.container.js [hunk 1]
```diff
@@ -4,16 +4,33 @@ import { compose } from 'recompose'
...
+  getSelectedAddress,
 } from '../../../selectors/selectors'
+import { getOriginFromUrl } from '../../../helpers/utils/util'
 
 const mapStateToProps = state => {
+  const addressConnectedToCurrentTab = getAddressConnectedToCurrentTab(state)
+  const { title, url } = state.appState.currentActiveTab || {}
+  let tabToConnect
+
+  if (!addressConnectedToCurrentTab && url && !url.match(/^chrome-extension:\/\/.+\/home.html#connected$/)) {
```
Comment (Gudahtt) [inline ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: This `chrome-extension` protocol is only used on Chrome. Firefox uses `moz-extension`, and other browsers probably use others.

Maybe we could use an allow list here instead? Something like this:
```suggestion
  if (!addressConnectedToCurrentTab && url && ['http', 'https'].includes((new URL(url)).protocol)) {
```

The last time we did something like this, here's the list of dapp protocols we accepted: `['http:', 'https:', 'dweb:', 'ipfs:', 'ipns:', 'ssb:']`. Though in practice it's possible only `http` and `https` are used with MetaMask now.

Comment (danjm) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: right...

Comment (danjm) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: maybe there is a simpler solution that is future proof against possible additional protocols

Comment (Gudahtt) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: The "Connect page" idea would make an allow list no longer necessary. We could validate the field, add warnings, etc.

That protocol list was used when we first rolled out privacy mode by default and had a home notification prompting for the manual connect. But when we moved it to the connections settings page, the protocol list was removed and we left it up to the user to know what they're doing.

Comment (Gudahtt) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: We could also look for our extension specifically (using `extension.runtime.id`). But then it still allows connecting to things like files, pictures, etc. which is also undesirable.

Comment (danjm) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: @Gudahtt What do you think about e58a2821a?

Comment (Gudahtt) [reply ui/app/components/app/connected-sites-list/connected-sites-list.container.js]: Sure, that works at least for ignoring _our_ UI. For the other cases where the button isn't desired, I guess it's fine to leave it up to the user to know what they're doing 🤷‍♂. This is probably at least as good of a solution as I suggested.

**Review by Gudahtt — APPROVED** [review]
Looks good to me!

**Pull request merged**